bin/yb-ctl

```diff
@@ -1022,48 +1072,120 @@ class ClusterControl:
-            time.sleep(SLEEP_TIME_IN_SEC)
-        logging.error("Failed waiting for {} tservers, got {}".format(
-            num_alive_ts, num_yb_admin_ts))
-        return False
+
+        logging.info("Waiting for master leader election tablet server registration.")
+        if not retry_fn_with_timeout(call_check_for_ts, self.options.timeout):
+            logging.error("Failed waiting for {} tservers, got {}".format(
+                num_alive_ts, num_yb_admin_ts))
+            return False
+        return True
 
     def wait_for_cluster_or_raise(self):
         if not self.wait_for_cluster():
             raise RuntimeError("Timed out waiting for Yugabyte cluster!")
 
-    def show_node_status(self, daemon_id):
-        pid = self.get_pid(daemon_id)
+    def _print_status_box(self, title, body_kv_list):
+        print("-" * 100)
+        print("| {:<96} |".format(title))
+        print("-" * 100)
+        for k, v in body_kv_list:
+            print("| {:20}: {:<74} |".format(k, v))
+        print("-" * 100)
 
-        def get_address(port_type):
-            return self.options.get_address(daemon_id, port_type, include_base_url=True)
-
-        if pid is None:
-            logging.info("Server {} is not running".format(daemon_id))
-        else:
-            info_list = [
-                ("type", daemon_id.daemon_type),
-                ("node_id", daemon_id.index),
-                ("PID", pid),
-                ("admin service", "http://" + get_address("http"))
-            ]
+    def _cluster_status(self):
+        cluster_uuid = None
+        try:
+            with open(os.devnull, "wb") as devnull:
+                cluster_uuid = subprocess.check_output(
+                    "curl localhost:7000/cluster-config | grep cluster_uuid",
+                    shell=True,
+                    stderr=devnull).strip()
+            regex_groups = re.match('cluster_uuid: "(.*)"', cluster_uuid).groups()
+            # We will display None if the subprocess fails...
+            cluster_uuid = regex_groups[0] if regex_groups else None
+        except subprocess.CalledProcessError as e:
+            pass
 
-            if daemon_id.is_tserver():
-                info_list.extend([
-                    ("cql service", get_address("cql_rpc")),
-                    ("redis service", get_address("redis_rpc")),
+        title = "Cluster UUID: {}".format(cluster_uuid)
+        server_counts = self.get_number_of_servers_map()
+        info_kv_list = [
+            ("Master Count", server_counts.get(DAEMON_TYPE_MASTER)),
+            ("TServer Count", server_counts.get(DAEMON_TYPE_TSERVER)),
+            ("Replication Factor", self.get_replication_factor()),
+            # TODO: this might cause issues if the first master is down...
+            ("Web UI", "http://{}/".format(
+                self.options.get_address(DaemonId(DAEMON_TYPE_MASTER, 1), "http", True))),
+            ("Cluster Data", self.options.cluster_base_dir)
```

Comment:
I think we should show YSQL, JDBC etc. info for 1 node on cluster creation as well. It doesn't present a complete picture of our system otherwise.

Reply:
Honestly, I would even consider printing the whole status (so cluster + all node blocks), but I think Karthik felt that would be too cluttered...I'm not sure if printing just 1 node would be ideal though, as that might be confusing if we have a mode where we print all the nodes and at startup we just print 1?

cc @rkarthik007 @schoudhury

Reply:
I didn't mean print details for one node. What I meant is print YSQL, JDBC etc. with the IP of first node. That way, users know how to connect to our DB. For a common single node install, it's anyway the only way to connect. So, output of create will be something like this:
```
./bin/yb-ctl --data_dir /tmp/yugabyte-local-cluster create
----------------------------------------------------------------------------------------------------
| Cluster UUID: 4c33a667-a085-4182-a27c-be94462afef0                                               |
----------------------------------------------------------------------------------------------------
| Master Count        : 1                                                                          |
| TServer Count       : 1                                                                          |
| Replication Factor  : 1                                                                          |
| Web UI              : http://127.0.0.1:7000/                                                     |
| Cluster Data        : /tmp/yugabyte-local-cluster                                                |
| JDBC                : postgresql://postgres@127.0.0.1:5433                                       |
| YSQL                : ./bin/psql -U postgres -h 127.0.0.1 -p 5433                                |
| YCQL                : ./bin/cqlsh 127.0.0.1 9042                                                 |
| YEDIS               : ./bin/redis-cli -h 127.0.0.1 -p 6379                                       |
----------------------------------------------------------------------------------------------------
```